backend/server/routers/courses.py

```diff
@@ -322,5 +393,56 @@ def coursesUnlockedWhenTaken(userData: UserData, courseToBeTaken: str):
     }
 
 def unlocked_set(courses_state):
-    """fetch the set of unlocked courses from the courses_state of a getAllUnlocked call"""
+    """ Fetch the set of unlocked courses from the courses_state of a getAllUnlocked call """
     return set(course for course in courses_state if courses_state[course]['unlocked'])
+
+
+def fuzzy_match(course: tuple, search_term: str):
+    """ Gives the course a weighting based on the relevance to the search """
+    (code, title) = course
+
+    # either match against a course code, or match many words against the title
+    # (not necessarily in the same order as the title)
+    search_term = search_term.lower()
+    if re.match('[a-z]{4}[0-9]', search_term):
+        return fuzz.ratio(code.lower(), search_term)
+
+    return max(fuzz.ratio(code.lower(), search_term),
+               sum([fuzz.partial_ratio(title.lower(), word)
+                       for word in search_term.split(' ')]))
+
+def weight_course(course: tuple, search_term: str, structure: dict,
+                  major_code: Optional[str] = None, minor_code: Optional[str] = None):
+    """ Gives the course a weighting based on the relevance to the user's degree """
+    weight = fuzzy_match(course, search_term)
+    (code, title) = course
+
+    # TODO: integrate electives weighting
```

Comment:
Does this comment still matter?

Reply:
To a degree, we could consider weighting electives differently to core courses but it's probably not important enough for a comment